src/cart/crt_init.c

```diff
@@ -472,13 +460,19 @@ crt_init_opt(crt_group_id_t grpid, uint32_t flags, crt_init_options_t *opt)
 
 		if (prov == CRT_NA_OFI_VERBS_RXM ||
 		    prov == CRT_NA_OFI_TCP_RXM) {
-			char *srx_env;
+			char *env;
 
-			srx_env = getenv("FI_OFI_RXM_USE_SRX");
-			if (srx_env == NULL) {
-				D_WARN("FI_OFI_RXM_USE_SRX not set, set=1\n");
+			env = getenv("FI_OFI_RXM_USE_SRX");
+			if (env == NULL) {
+				D_INFO("FI_OFI_RXM_USE_SRX not set, set=1\n");
 				setenv("FI_OFI_RXM_USE_SRX", "1", true);
 			}
+
+			env = getenv("FI_OFI_RXM_ENABLE_DIRECT_SEND");
+			if (env == 0) {
+				D_INFO("FI_OFI_RXM_ENABLE_DIRECT_SEND was not set; set=1\n");
+				setenv("FI_OFI_RXM_ENABLE_DIRECT_SEND", "1", true);
+			}
```

Comment:
what is the plan on dealing with mpi? wont this cause problems if mpi does not pre-register their buffers?

Reply:
setenv() sets environment variables with process scope, so assume one daos client process call it will not affect another MPI process?

Reply:
the MPI process is the same as the DAOS client process

Reply:
Yes as Mohamad said, its the same process that does mpi communication and daos communication

Reply:
I am not sure what is the concern with MPI. If MPI registers memory, then they will go through the same process as us, if not, they will go through bounce buffer.

Reply:
Seems ofi does not internally check whether or not the provided buffer is registered for not (rxm_send_eager()).
Mercury's RPC input/output buffer and bulk buffer all registered, so normal DAOS workload should be fine.
But yes seems there is a problem that if the MPI program linked with libdaos, and the MPI program itself also use RXM provider but don't register its local buffer ahead, that possibly with problem.
We may provide an init option flag to crt_init() and daos_init() if that is a concern?